compass/landice/tests/humboldt/mesh.py

```diff
@@ -0,0 +1,361 @@
+
+        print('Marking domain boundaries dirichlet')
+        args = ['mark_domain_boundaries_dirichlet.py',
+                '-f', 'Humboldt_1to10km.nc']
+        check_call(args, logger=logger)
+
+        print('calling set_lat_lon_fields_in_planar_grid.py')
+        args = ['set_lat_lon_fields_in_planar_grid.py', '-f',
+                'Humboldt_1to10km.nc', '-p', 'gis-gimp']
+        check_call(args, logger=logger)
+
+        print('creating graph.info')
+        make_graph_file(mesh_filename='Humboldt_1to10km.nc',
+                        graph_filename='graph.info')
+
+    def build_cell_width(self):
+        """
+        Determine MPAS mesh cell size based on user-defined density function.
+
+        This includes hard-coded definition of the extent of the regional
+        mesh, a flood-fill routine to ignore glaciers and ice caps not
+        connected to the ice sheet, and user-defined mesh density functions
+        based on observed flow speed and distance to the ice margin. In the
+        future, this function and its components will likely be separated
+        into separate generalized functions to be reusable by multiple test
+        groups.
+        """
+        # get needed fields from GIS dataset
+        f = netCDF4.Dataset('greenland_8km_2020_04_20.epsg3413.nc', 'r')
+        f.set_auto_mask(False)  # disable masked arrays
+
+        x1 = f.variables['x1'][:]
+        y1 = f.variables['y1'][:]
+        thk = f.variables['thk'][0, :, :]
+        topg = f.variables['topg'][0, :, :]
+        vx = f.variables['vx'][0, :, :]
+        vy = f.variables['vy'][0, :, :]
+
+        dx = x1[1] - x1[0]  # assumed constant and equal in x and y
+        nx = len(x1)
+        ny = len(y1)
+
+        sz = thk.shape
+
+        # Define extent of region to mesh.
+        # These coords are specific to the Humboldt Glacier mesh.
+        xx0 = -630000
+        xx1 = 84000
+        yy0 = -1560000
+        yy1 = -860000
+        geom_points = np.array([  # list of xy "node" coordinates
+            ((xx0, yy0), 0),
+            ((xx1, yy0), 0),
+            ((xx1, yy1), 0),
+            ((xx0, yy1), 0)],
+            dtype=jigsawpy.jigsaw_msh_t.VERT2_t)
+
+        geom_edges = np.array([    # list of "edges" between nodes
+            ((0, 1), 0),
+            ((1, 2), 0),
+            ((2, 3), 0),
+            ((3, 0), 0)],
+            dtype=jigsawpy.jigsaw_msh_t.EDGE2_t)
+
+        # flood fill to remove island, icecaps, etc.
+        searchedMask = np.zeros(sz)
+        floodMask = np.zeros(sz)
+        iStart = sz[0] // 2
+        jStart = sz[1] // 2
+        floodMask[iStart, jStart] = 1
+
+        neighbors = np.array([[1, 0], [-1, 0], [0, 1], [0, -1]])
+
+        lastSearchList = np.ravel_multi_index([[iStart], [jStart]],
+                                              sz, order='F')
+
+        # flood fill -------------------
+        # In the future this might become a separate function
+        # to be used by multiple test groups.
```

Comment:
I would definitely be in favor of pulling this out into the `compass.landice` framework in the future.